### ISSUE TITLE
Overload resolution doesn't work with collection expressions

### ISSUE THREAD

**Issue by thinker227** [dotnet/roslyn#69962]
**Version Used**: 4.8.0-3.23455.11 (b5e47b87)

**Steps to Reproduce**:

1. Create a method with two overloads which take two different `IEnumerable<T>`s as their parameters.
```cs
void Foo(IEnumerable<int> xs) {}
void Foo(IEnumerable<string> xs) {}
```
2. Call the method with a collection expression as the argument.
```cs
Foo([1, 2, 3]);
```

[Sharplab link](https://sharplab.io/#v2:EYLgtghglgdgNAFxANwKYCcoDMCecAmIA1AD4ACADAARkCMALANwCwAUG2QExUDCbA3myrCa9KgDEA9pIAUdAMwAeWAgB8VAB4BnAJRV+AXyEiyYqbIWK6Fddr2G2x4aaoAhCOhk6n+nyInSMgDatHBUnGHyALo6LKz+/ubBAEQQyWHJwOlUyQDGyTFx/kasBkA=)

**Diagnostic Id**: CS0121

**Expected Behavior**: The correct overload is chosen based on the inferred type of the collection expression (eg. `IEnumerable<int>` in the above example, so `void Foo(IEnumerable<int>)` is chosen).

**Actual Behavior**: CS0121 (`The call is ambiguous between the following methods or proper
ties: 'C.Foo(IEnumerable<int>)' and 'C.Foo(IEnumerable<string>)'`) is reported when calling the method. The error is *not* reported in-IDE (VSCode with C# Dev Kit) but is reported when using `dotnet build`.

**Comment by CyrusNajmabadi** [comment]
@cston @jcouv already fixed?

**Comment by thinker227** [comment]
Odd, seems like `dotnet build` is using 4.8.0-1.23374.10 (520e2f10) while Dev Kit/Omnisharp is using 4.8.0-3.23455.11 (b5e47b8).

Well, nice that it's fixed. I'll download SDK 8.0.0-rc.1 and try that.
